### PR TITLE
Auto puge for WooCommerce Advanced Bulk Edit plugin.

### PR DESCRIPTION
**Pull request by stanleybz**
[Asana] Cache is not automatically cleared with woocommerece plugin
https://www.litespeedtech.com/support/forum/threads/cache-is-not-automatically-cleared.17426/

admin-ajax form data sample:
```
type: saveproducts
nonce: afdc96369c
isfirst: 1
data[_regular_price]: 464$###0$###1#^#463$###0$###3#^#462$###0$###5#^#456$###0$###6#^#311$###0$###4
data[_sale_price]: 464$###0$###2#^#463$###0$###4#^#462$###0$###4
columns[_product_permalink]: 170
columns[product_type]: 100
columns[_visibility]: 190
columns[_stock_status]: 80
columns[_stock]: 80
columns[_tax_status]: 80
columns[_featured]: 80
columns[post_status]: 70
columns[post_content]: 250
columns[_sku]: 60
columns[_sale_price]: 80
columns[_regular_price]: 80
columns[product_cat]: 150
columns[_product_image_gallery]: 90
columns[_thumbnail_id]: 80
columns[post_title]: 250
columns[ID]: 60
columns[sel]: 30
filters: 
```

### CODE REVIEW

File: litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php [hunk 1]
```diff
@@ -962,5 +963,54 @@ public static function get_instance()
...
+		* admin-ajax form-data structure
+		* array(
+		*		"type" => "saveproducts",
+		*		"data" => array(
+		*			"column1" => "464$###0$###2#^#463$###0$###4#^#462$###0$###6#^#",
+		*			"column2" => "464$###0$###2#^#463$###0$###4#^#462$###0$###6#^#"
+		*		)
+		*	)
+		*/
+		$stock_string_data = $_POST[ 'data' ] ;
```
Comment (hailite) [inline litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php]: If `$stock_string_data` is only used once, no reason to set this variable.

File: litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php [hunk 2]
```diff
@@ -962,5 +963,54 @@ public static function get_instance()
...
+		* @since 2.9.8.4
+		*/
+		add_action( 'wp_ajax_wpmelon_adv_bulk_edit', 'LiteSpeed_Cache_ThirdParty_WooCommerce::bulk_edit_purge', 1 ) ;
+	}
+
+	public static function bulk_edit_purge()
+	{
+		$type = $_POST[ 'type' ] ;
+
+		if ( $type !== 'saveproducts' && empty( $_POST[ 'data' ] ) ) return ;
```
Comment (hailite) [inline litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php]: Should here be `&& or `||`?

File: litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php [hunk 3]
```diff
@@ -962,5 +963,54 @@ public static function get_instance()
...
+		*		"type" => "saveproducts",
+		*		"data" => array(
+		*			"column1" => "464$###0$###2#^#463$###0$###4#^#462$###0$###6#^#",
+		*			"column2" => "464$###0$###2#^#463$###0$###4#^#462$###0$###6#^#"
+		*		)
+		*	)
+		*/
+		$stock_string_data = $_POST[ 'data' ] ;
+		$stock_string_arr = array() ;
+		foreach ( $stock_string_data as $stock_key => $stock_value ) {
```
Comment (hailite) [inline litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php]: Is `$stock_key` ever used?

File: litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php [hunk 4]
```diff
@@ -962,5 +963,54 @@ public static function get_instance()
...
+			$stock_string_arr = array_merge( $stock_string_arr, explode( '#^#', $stock_value ) ) ;
+		}
+
+		$lscwp_3rd_woocommerce = new LiteSpeed_Cache_ThirdParty_WooCommerce ;
+
+		if ( sizeof( $stock_string_arr ) < 1 ) return ;
```
Comment (hailite) [inline litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php]: Please use `count()` instead.

File: litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php [hunk 5]
```diff
@@ -962,5 +963,54 @@ public static function get_instance()
...
+		* Auto puge for WooCommerce Advanced Bulk Edit plugin,
+		* Bulk edit hook need to add to preload as it will die before detect.
+		* @since 2.9.8.4
+		*/
+		add_action( 'wp_ajax_wpmelon_adv_bulk_edit', 'LiteSpeed_Cache_ThirdParty_WooCommerce::bulk_edit_purge', 1 ) ;
+	}
+
+	public static function bulk_edit_purge()
+	{
+		$type = $_POST[ 'type' ] ;
```
Comment (hailite) [inline litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php]: If `$type` is only used once, no reason to set it.

**Comment by stanleybz** [comment]
@hailite Updated.

**Comment by hailite** [comment]
Please check https://github.com/litespeedtech/lscache_wp/commit/a0dba4c9f64065db9c9a7fd03d17c0a4bdd130c2

**Pull request closed without merging**